roachtest: refactor github post create tests as datadriven

DarrylWong opened the pull request:
The test now renders the issues in markdown.

Release note: none
Fixes: none
Epic: none

DarrylWong committed:
roachtest: refactor github post create tests as datadriven

The test now renders the issues in markdown.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/132795)
<!-- Reviewable:end -->

DarrylWong committed:
github: add matching blank line to issue posting

This change adds a matching blank line to a <\sub> tag in
github issue posting. Without this, the subscript won't be
properly closed and following text will be affected.

herkolategan commented on pkg/cmd/bazci/githubpost/issues/formatter_unit.go:
```diff
@@ -155,7 +155,7 @@ func (unitTestFormatterTyp) Body(r *Renderer, data TemplateData) error {
 			"https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues",
 		)
-		r.Escaped("\n")
+		r.Escaped("\n\n") // need a blank line here since we have one above
```
Interesting, I see all our issues have the Jira issue displaying as part of the sub! Nice catch!

herkolategan reviewed (approved):
Nice work!
